### ISSUE TITLE
Add new annotation to specify enum value in serialized json

### ISSUE THREAD

**Issue by ranquild** [dart-lang/json_serializable#38]
There is case when enum name in dart and in json does not match, it would be nice to have construction like this:

```
enum FieldType {
  @JsonEnum('Text)
  text,
  @JsonEnum('Number')
  number
}
```

to solve those issues.

**Comment by kevmoo** [comment]
Pull requests welcome!

**Comment by kevmoo** [comment]
Looks like we're blocked on this https://github.com/dart-lang/sdk/issues/30934

We might be able to do something ~ugly and have a mapping on the enum definition – WDYT?

**Comment by lrhn** [comment]
The spec currently doesn't allow annotations on enum item declarations. That's is probably an oversight, so we should just fix it.

**Comment by ranquild** [comment]
@kevmoo We would better wait for dart supporting enum annotations.

**Comment by kevmoo** [comment]
No longer blocked – I think!

**Comment by kevmoo** [comment]
Seems like we're blocked by https://github.com/dart-lang/sdk/issues/33373

**Comment by kevmoo** [comment]
No longer blocked. Will require latest pkg:analyzer and SDK, though.

**Comment by sir-boformer** [comment]
It would be great if arbitrary integer values would also be supported:

```dart
enum HttpStatus {
  @JsonEnum(200)
  success,
  @JsonEnum(404)
  notFound
}
```

**Comment by kevmoo** [comment]
> It would be great if arbitrary integer values would also be supported:

Good idea! Noted...

**Comment by kevmoo** [comment]
Note – PR out for this https://github.com/dart-lang/json_serializable/pull/251

Comments welcome. Still discussing if we want a new annotation class or if we should just use `JsonKey`...
